### PR TITLE
[Bugfix][v1] Fix step pooler implementation and step pooling usage in v1

### PR DESCRIPTION
**Pull request by Isotr0py**
## Essential Elements of an Effective PR Description Checklist
- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.

## Purpose
- Fix redundant `pooled_data = []` in step pooler implementation
- Fix missing `prompt_token_ids` in pooling metadata for step pooling when using v1 runner

## Test Plan
```
pytest -s -v tests/models/language/pooling/test_reward.py
```

## Test Result
- Test should pass on both v0 and v1 engine

### CODE REVIEW

**Comment by Isotr0py** [comment]
Also cc @maxdebayser

File: tests/conftest.py [hunk 1]
```diff
@@ -1042,6 +1042,10 @@ def encode(self,
         req_outputs = self.model.embed(inputs, *args, **kwargs)
```
Comment (DarkLight1337) [inline tests/conftest.py:1042]: I think we should actually update `encode` to match the `LLM` API so it can be used for reward models, while we introduce `embed` specifically for embedding models

**Review by DarkLight1337 — APPROVED** [review]

File: vllm/v1/worker/gpu_model_runner.py [hunk 2]
```diff
@@ -1708,6 +1708,9 @@ def load_model(self) -> None:
                 )
                 model_loader.load_weights(self.model,
                                           model_config=self.model_config)
+            if (self.is_pooling_model
+                    and hasattr(self.model._pooler, "step_tag_id")):
+                self.input_batch.sampling_needs_token_ids = True
```
Comment (maxdebayser) [inline vllm/v1/worker/gpu_model_runner.py]: Can we move this check to `interfaces.py`/`registry.py`? Some sparse embedding models also require the prompt token IDs and perhaps this list will grow in the future.

File: vllm/v1/worker/gpu_input_batch.py [hunk 3]
```diff
@@ -59,21 +59,23 @@ def get_token_id(self, idx: int) -> int:
...
+        block_sizes: list[int],  # The block_size of each kv cache group
+        sampling_needs_token_ids: bool = False,
     ):
         self.max_num_reqs = max_num_reqs
         self.max_model_len = max_model_len
         self.max_num_batched_tokens = max_num_batched_tokens
         self.device = device
         self.pin_memory = pin_memory
         self.vocab_size = vocab_size
+        self.sampling_needs_token_ids = sampling_needs_token_ids
```
Comment (maxdebayser) [inline vllm/v1/worker/gpu_input_batch.py]: I know that it's had to come up with good names, but can we rename this to something that doesn't imply that it's exclusive for sampling? Something along the lines of `logits_processing_needs_token_ids`.

**Review by maxdebayser — CHANGES_REQUESTED** [review]
Thanks for the fix! Overall it looks good to me. I've only requested a few small refactorings.

**Review by maxdebayser — APPROVED** [review]
LGTM, thanks!
